c/shared/source/tx_shared/tx_shared.c

```diff
@@ -405,6 +405,31 @@ static size_t stm_read(ctlStreamCallbacks *cb, void *stream, char **ptr) {
     return 0; /* Suppress compiler warning */
 }
 
+static size_t stm_xml_read(ctlStreamCallbacks *cb, void *stream, char **ptr, xmlDocPtr *doc){
+    int res, size = BUFSIZ;
+    char chars[BUFSIZ];
+    xmlParserCtxtPtr ctxt;
+
+    Stream *s = stream;
+    txCtx h = cb->direct_ctx;
+    if (h->seg.refill != NULL)
+        return h->seg.refill(h, ptr);
+
+    *ptr = s->buf;
```

Comment:
Not sure `ptr` will point to something sensible after we go though this loop (given that `buf` gets re-used), and if that's right its concerning that we're passing it back from here.

Reply:
I looked into this a bit, but not sure if I understood. Is it the `return h->seg.refill(h, ptr); `line that's concerning, or the `*ptr = s->buf;`?

Reply:
The latter. This could easily be something I just don't understand. What is `ptr` supposed to represent as a return parameter?